NETProvider/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbParameterCollection.cs

```diff
@@ -90,6 +91,21 @@ public override object SyncRoot
 			get { return ((ICollection)_parameters).SyncRoot; }
 		}
 
+		internal bool? _collectionHasParameterWithUnicodeName;
```

Comment:
I feel that checking that field is too much. For instance if we recompute the value with every change it will still be correct but the test will assume it's not.